src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptCompletionServiceWithProviders.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Completion;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
+{
+    internal abstract class VSTypeScriptCompletionServiceWithProviders : CompletionServiceWithProviders
+    {
+        internal VSTypeScriptCompletionServiceWithProviders(Workspace workspace)
+            : base(workspace)
+        {
+        }
```

Comment:
will we have to stage this in increments to have TS move to this?  or will things still work as they have IVT, and we'll then ask them to move to this asap once they update to this version?  also, we should likely pull in Andrew Casey and Will Smith (intentionally not tagging them yet) once we think we have the right final shape to make sure they are aware of this.

Reply:
I'll send PRs to F# and TS.

Reply:
> will we have to stage this in increments to have TS move to this? 

Yes, first we add this API and then we move them over.